stdlib-integration-tests/io/tests/io_socket_tests.bal

```diff
@@ -0,0 +1,426 @@
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/io;
+import ballerina/test;
+import ballerina/socket;
+import ballerina/log;
+import ballerina/lang.'string as langstring;
+
+@test:Config {}
+public function testSocketChannelReadBytes() {
+    socket:Client socketClient = new ({
+        host: "localhost",
+        port: 61599
+    });
+    string payload = "Hello Ballerina World!!!!";
+    socketWrite(socketClient, payload);
+
+    var result = socketClient->read();
+    if (result is [byte[], int]) {
+        var [reply, length] = result;
+        if (length > 0) {
+            var byteChannel = io:createReadableChannel(reply);
+            if (byteChannel is io:ReadableByteChannel) {
+                var content = io:channelReadBytes(byteChannel);
+                if (content is byte[]) {
+                    test:assertEquals(content, payload.toBytes());
+                } else {
+                    test:assertFail(msg = content.message());
+                }
+            } else {
+                test:assertFail(msg = byteChannel.message());
+            }
+        }
+    } else {
+        test:assertFail(msg = result.message());
+    }
+
+    var closeResult = socketClient->close();
+    if (closeResult is error) {
+        test:assertFail(msg = closeResult.message());
+    }
+}
+
+@test:Config {dependsOn: ["testSocketChannelReadBytes"]}
+public function testSocketChannelReadBytesAsStream() {
+    socket:Client socketClient = new ({
+        host: "localhost",
+        port: 61599
+    });
+    string payload = "Hello Ballerina World Again!!!!";
+    socketWrite(socketClient, payload);
+    byte[] byteArr = [];
+
+    var result = socketClient->read();
+    if (result is [byte[], int]) {
+        var [reply, length] = result;
+        if (length > 0) {
+            var byteChannel = io:createReadableChannel(reply);
+            if (byteChannel is io:ReadableByteChannel) {
+                var content = io:channelReadBlocksAsStream(byteChannel, 1);
+                if (content is stream<io:Block>) {
+                    _ = content.forEach(function(io:Block val) {
+                                           foreach byte b in val {
+                                               byteArr.push(b);
+                                           }
+                                       });
+                    string|error returnedString = langstring:fromBytes(byteArr);
```

Comment:
Doesnt byteArr.cloneWithType(string) work?

Reply:
`cloneWithType` generally takes a value and converts it to a different target type of the same basic type (e.g., mapping basic type - map to record and vice-versa, list basic type - array to tuple and vice-versa). The exception (converting from one basic type to a different basic type) is with numeric values.

`fromBytes` is a function introduced specifically to "construct a string from its UTF-8 representation byte[]", so that's the correct function to use here.

Reply:
It gives a type conversion error.
```
{ballerina/lang.typedesc}ConversionError
```

Reply:
@MaryamZi Ack